internal/commands/pull/command.go

```diff
@@ -60,23 +62,47 @@ func (cmd *Command) Handler(profile *user.Profile, ui terminal.UI, clients cli.C
 		return err
 	}
 
-	pathTarget, zipPkg, err := cmd.doExport(profile, clients.Realm, app.GroupID, app.ID)
+	clientZipPkgs, err := cmd.inputs.resolveClient(ui, clients.Realm, app.GroupID, app.ID)
 	if err != nil {
 		return err
 	}
 
-	pathRelative, err := filepath.Rel(profile.WorkingDirectory, pathTarget)
+	pathTarget, zipPkg, err := cmd.doExport(profile, clients.Realm, app.GroupID, app.ID)
 	if err != nil {
 		return err
 	}
 
-	proceed, err := checkAppDestination(ui, pathTarget)
+	if clientZipPkgs != nil {
+		templatePath := filepath.Join(pathTarget, local.FrontendPath)
+		if proceed, err := checkPathDestination(ui, templatePath); err != nil {
+			return err
+		} else if !proceed {
+			return nil
+		}
+
+		for templateID, templateZipPkg := range clientZipPkgs {
+			if err := local.WriteZip(templatePath, templateZipPkg); err != nil {
+				return err
+			}
+			ui.Print(terminal.NewTextLog(fmt.Sprintf("Saved template %s to disk", templateID)))
+		}
+
+		pathTarget = filepath.Join(pathTarget, local.BackendPath)
+	}
+
+	// App path
+	proceed, err := checkPathDestination(ui, pathTarget)
```

Comment:
if you go with the suggested change above, make sure to now use `pathBackend` here

also, we end up using the same confirm prompt for both backend and frontend templates, and the messaging only changes with the filepath shown.  does it make sense to adjust the language on each to make it clear when we're writing the "backend stuff" or "frontend stuff" (I would imagine that would also only really want to happen if templates were involved in the first place)

if so, i think breaking out `checkPathDestination` to two funcs, one for `checkBackend` and one for `checkFrontend` would be a good idea here.  and `checkBackend` could then either use the existing code for when templates arent involved, or mention that this would be writing "backend stuff"...

Reply:
I think it's fine to leave it as is, since if there is no template, then there is no concept of `frontend` or `backend` paths

Reply:
feel free to settle on namings how you see fit, but what I don't love with the current implementation is that when templates are involved, you are mutating the value of `pathTarget`, and the order of operations matters significantly within that if block.  fwiw, i'm totally fine with us calling it "backend" even when a template isn't involved

that is why I want us to use 3 `var`s here for "app" (or could call it "root", as it effectively the same as `local.App.RootDir`), "backend" (which _is_ just the "app" when templates aren't involved, so maybe we call that the `pathApp`), and then "frontend" (which is where we write one or many client templates)

if/when we go down that route, we just need to be very careful we are then using the right versions of these 3 `var`s (and for what we use to derive `pathRelative`) to keep things working as expected